src/components/implementation/no_interface/vkernel/vk_server.c

```diff
@@ -0,0 +1,47 @@
+
+#include "vk_types.h"
+
+int
+vkernel_find_vm(thdid_t tid)
+{
+	int i;
+
+	for (i = 0 ; i < VM_COUNT ; i ++) {
+		if ((vmx_info[i].inithd)->thdid == tid) break;
+	}
+	assert (i < VM_COUNT);
+
+	return i;
+}
+
+int
+vkernel_server(int a, int b, int c)
```

Comment:
This is the INV function provided by Vkernel to VMs, so named it vk_server, that sounded appropriate to me.